core/DataTable/Renderer/Csv.php

```diff
@@ -242,7 +242,12 @@ public function formatValue($value)
         if (is_string($value)) {
             $value = str_replace(["\t"], ' ', $value);
 
-            if (strpos($value, '"') !== false || strpos($value, $this->separator) !== false) {
+            // surround value with double quotes if it contains a double quote or a commonly used separator
+            if (strpos($value, '"') !== false
+                || strpos($value, $this->separator) !== false
+                || strpos($value, ',') !== false
+                || strpos($value, ';') !== false
+            ) {
```

Comment:
Should we expand the already existing `testCSVRendererCorrectlyEscapesHeadersAndValues` to use all four variants (quote, comma, semicolon, a different `setSeparator` call value)?

Reply:
sure. won't hurt :-)